benchmarks/functional_autograd_benchmark/audio_text_models.py

```diff
@@ -0,0 +1,128 @@
+import torch
+import torch.nn as nn
+
+import torchaudio_models as models
+
+from utils import make_functional, load_weights
+
+def get_wav2letter(device):
+    N = 10
+    input_frames = 700
+    vocab_size = 28
+    model = models.Wav2Letter(num_classes=vocab_size)
+    criterion = torch.nn.NLLLoss()
+    model.to(device)
+    params, names = make_functional(model)
+
+    inputs = torch.rand([N, 1, input_frames], device=device)
+    labels = torch.rand(N, 3, device=device).mul(vocab_size).long()
+
+    def forward(*new_params):
+        load_weights(model, names, new_params)
+        out = model(inputs)
+
+        loss = criterion(out, labels)
+        return loss
+
+    return forward, params
+
+def get_deepspeech(device):
+    supported_rnns = {
+        'lstm': nn.LSTM,
+        'rnn': nn.RNN,
+        'gru': nn.GRU
+    }
+    sample_rate = 16000
+    window_size = 0.02
+    window = "hamming"
+    audio_conf = dict(sample_rate=sample_rate,
+                      window_size=window_size,
+                      window=window,
+                      noise_dir=None)
+
+
+    N = 10
+    num_classes = 10
+    spectrogram_size = 161
+    # Commented are the original sizes in the code
+    seq_length = 500  # 1343
+    target_length = 10  # 50
+    labels = torch.rand(num_classes, device=device)
+    inputs = torch.rand(N, 1, spectrogram_size, seq_length, device=device)
+    # Sequence length for each input
+    inputs_sizes = torch.rand(N, device=device).mul(seq_length * 0.1).add(seq_length * 0.8)
+    targets = torch.rand(N, target_length, device=device)
+    targets_sizes = torch.full((N,), target_length, dtype=torch.int, device=device)
+
+    model = models.DeepSpeech(rnn_type=supported_rnns["lstm"], labels=labels, rnn_hidden_size=1024, nb_layers=5,
+                              audio_conf=audio_conf, bidirectional=True)
+    model = model.to(device)
+    criterion = nn.CTCLoss()
+    params, names = make_functional(model)
+
+    def forward(*new_params):
+        load_weights(model, names, new_params)
+        out, out_sizes = model(inputs, inputs_sizes)
+        out = out.transpose(0, 1)  # For ctc loss
+
+        loss = criterion(out, targets, out_sizes, targets_sizes)
+        return loss
+
+    return forward, params
+
+def get_transformer(device):
+    # For most SOTA research, you would like to have embed to 712, nhead to 12, bsz to 64, tgt_len/src_len to 128.
+    N = 64
+    seq_length = 128
+    ntoken = 50
+    model = models.TransformerModel(ntoken=ntoken, ninp=720, nhead=12, nhid=2048, nlayers=2)
```

Comment:
Is `embed == 712` used here at all? Is that the same thing as `ninp` (is the 720 a typo)?